source/common/upstream/thread_aware_lb_impl.cc

```diff
@@ -185,5 +186,108 @@ LoadBalancerPtr ThreadAwareLoadBalancerBase::LoadBalancerFactoryImpl::create() {
   return lb;
 }
 
+bool ThreadAwareLoadBalancerBase::BoundedLoadHashingLoadBalancer::isHostOverloaded(
+    HostConstSharedPtr host, double weight) const {
+  /*
+  Consistent Hashing with Bounded Load
+  Ref: https://arxiv.org/abs/1608.01350
+  */
```

Comment:
nit: // comments. Also, can you briefly explain the algorithm to avoid having to read the entire paper?

Reply:
Explained the algorithm in method ThreadAwareLoadBalancerBase::BoundedLoadHashingLoadBalancer::chooseHost